### PR TITLE
Created the start of the chapter for the basic url design.

### PR DESCRIPTION
**Pull request by MrHus**
The chapter has a cheatsheet showing the basics of a REST resource
including the URL and HTTP methods.

Then "Resource" is defined and the various HTTP methods are
fleshed out. Each HTTP request method has its own heading and
abstract and concrete example.

### CODE REVIEW

File: 02-basic-url-design.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+| URL           | method | response         |
+| ------------- |--------| -----------------|
+| /cars         | GET    | A list of cars   |
+| /cars/1       | GET    | A single car     |
+| /cars/1       | POST   | The created car  |  
+| /cars/1       | PUT    | The updated car  |
+| /cars/1       | PATCH  | The updated car  |
+| /cars/1       | DELETE | The deleted car  |
+
+# Anatomy of a Resource.
```
Comment (Pouja) [inline 02-basic-url-design.md]: redundant period

**Comment by Pouja** [comment]
Maybe add a reference to the official w3 standards for the different methods? https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9.3

File: 02-basic-url-design.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+A REST API is a collection of resources. A Resource
+represents something in the domain of your application.
+For example: a domain for a webshop might have
```
Comment (Pouja) [inline 02-basic-url-design.md]: `might contain` is maybe better?

File: 02-basic-url-design.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+Stock, Product and Customer resources.
+
+A resource is something that can be manipulated.
+For example we can add, remove and update a resource.
+
+A resource can also be queried: we can get a list of
+a particular resource, but we can retrieve the details of
```
Comment (Pouja) [inline 02-basic-url-design.md]: `but we can also`

File: 02-basic-url-design.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+a specific resource.
+
+Before REST we would create specific URL's for each
+operation or query type. One URL to request a list,
```
Comment (Pouja) [inline 02-basic-url-design.md:36]: One URL to rule them all!

File: 02-basic-url-design.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+So what are these 'verbs'. 
+
+# Request Methods
+HTTP supports multiple 'requests methods' for a single
+URL. Request methods are the 'verbs' which specify what 
+you want to happen when you request that url.
+
+The Request Methods that REST commonly use are: GET, POST, PUT, 
+PATCH and DELETE. These are all specified in the HTTP standard and
```
Comment (Pouja) [inline 02-basic-url-design.md]: You can link here to https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html

File: 02-basic-url-design.md [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+act as a 'type' for a HTTP request.
+
+When you put a Request Method (verb) before a Resource something
+magical happens: You an probably guest what the request `DELETE /cars/1`
```
Comment (Pouja) [inline 02-basic-url-design.md]: `You can probably guess`

File: 02-basic-url-design.md [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+
+The list of resources is fetched using the following URL:
+
+`GET /{entity}`
+
+A singular resource is fetched via:
+
+`GET /{entity}/{identifier}`
+
+The relation ship between a singular and list looks the same
```
Comment (Pouja) [inline 02-basic-url-design.md]: `The relationship`

File: 02-basic-url-design.md [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+In REST we use the POST method to 'create' new resources. The URL
+for the POST request is the same as the list request. This is to
+denote that we are 'creating' a new resource which belongs to that
+list.
+
+The request is defined as:
+
+`POST /{entity}`
+
+For example, to create a new car we send:
```
Comment (Pouja) [inline 02-basic-url-design.md]: `we would send` sounds maybe better?

**Comment by Pouja** [comment]
:shipit:

File: 02-basic-url-design.md [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+a larger list of resources.
+
+The name of the entity is __always__ in plural. So if we have
+a Car resource the urls would become:
+
+`GET /cars` for the list of cars and `GET cars/1` for a single car.
+
+## POST
+The POST request method is what your browser uses when you
+submit a HTML form.
```
Comment (SanderBenschop) [inline 02-basic-url-design.md]: _an_ HTML form.

**Pull request merged**